Source/Urho3D/Graphics/OpenGL/OGLTexture.cpp

```diff
@@ -187,6 +187,31 @@ bool Texture::IsCompressed() const
            format_ == COMPRESSED_RGB_PVRTC_2BPPV1_IMG || format_ == COMPRESSED_RGBA_PVRTC_2BPPV1_IMG;
 }
 
+unsigned Texture::GetDataSize(int width, int height) const
+{
+    if (IsCompressed())
+    {
+        if (format_ == COMPRESSED_RGB_PVRTC_4BPPV1_IMG ||
+            format_ == COMPRESSED_RGBA_PVRTC_4BPPV1_IMG ||
+            format_ == COMPRESSED_RGB_PVRTC_2BPPV1_IMG ||
+            format_ == COMPRESSED_RGBA_PVRTC_2BPPV1_IMG)
+        {
+            const unsigned bpp = (format_ == COMPRESSED_RGB_PVRTC_2BPPV1_IMG ||
+                                format_ == COMPRESSED_RGBA_PVRTC_2BPPV1_IMG)?2:4;
+            
+            const unsigned xSize =  (bpp == 2) ? 8 : 4;
+            const unsigned blockSize = 8;
+            // For MBX don't allow the sizes to get too small
+            unsigned blocksWide = Max(2, width / xSize);
+            unsigned blocksHeight = Max(2, height / 4);
+            return blocksWide * blocksHeight * blockSize;
+        }
+        return GetRowDataSize(width) * ((height + 3u) >> 2u);
+    }
+    else
+        return GetRowDataSize(width) * height;
+}
+
```

Comment:
I am not entirely sure if moving `GetDataSize()` to subclasses is a good idea. In theory compression formats do not depend on graphics API, they depend on hardware. For example even though OpenGL can use PVRTC on iOS - this format is not available on androids. Also consider (do not laugh!) we add a metal backend. Both OpenGL and metal could use PVRTC so this code would have to be duplicated. So i think its best to have this logic in `Texture.cpp`. @eugeneko any thoughts?

Reply:
Format seems to be platform dependent. Would be nice if the texture format field would be not int but enum...

Reply:
Format is platform dependent, but code attached it to graphics API, not a platform.

Reply:
I mean format depends on graphics Api. It is exposed as int but the actual meaning depends on API you use.

Reply:
Not really. Both metal and opengl can use pvrtc, just like opengl and directx can use DXT5. Compressed formats depend on hardware, not graphics API.